ntfysh_client/NotificationDialog.cs

```diff
@@ -0,0 +1,263 @@
+                this.displayTimeoutTimer.Dispose();
+            }
+            if (this.updateTimer != null)
+            {
+                this.updateTimer.Stop();
+                this.updateTimer.Dispose();
+            }
+            if (timeout_ms > 0)
+            {
+                this.displayTimeoutTimer = new System.Timers.Timer(timeout_ms);
+                displayTimeoutTimer.Elapsed += handleTimeout;
+                this.displayTimeoutTimer.Start();
+
+                if (showTimeOutBar)
+                {
+                    this.progress = 100;
+                    this.updateTimer = new System.Windows.Forms.Timer();
+                    updateTimer.Interval = 100;
+                    this.updateTimer.Tick += this.UpdateProgress;
+                    this.updateTimer.Start();
+
+                    this.shownStopwatch = new Stopwatch();
+                    this.shownStopwatch.Start();
+
+                    this.progressBar1.Visible = true;
+                    this.lbTimeout.Visible = true;
+                    this._timeout = timeout_ms;
+                } else
+                {
+                    this.progressBar1.Visible = false;
+                    this.lbTimeout.Visible = false;
+                }
+            }
+            else
+            {
+                this.progressBar1.Visible = false;
+                this.lbTimeout.Visible = false;
+            }
+
+            // ok, show the window
+            this.Show();
+            this.SetWindowPosition();
+        }
+
+        private void UpdateProgress(object? sender, EventArgs e)
+        {
+            if (this.shownStopwatch == null)
+            {
+                return;
+            }
+            this.progress = (int)((this._timeout - this.shownStopwatch.ElapsedMilliseconds) * 100 / this._timeout);
+            this.lbTimeout.Text = $"{(int)(this._timeout - this.shownStopwatch.ElapsedMilliseconds) / 1000}";
+        }
+
+        protected override void SetVisibleCore(bool value)
+        {
+            this.SetWindowPosition();
+            if (value)
+            {
+                this.BringToFront();
+                AnimateWindow(
+                    this.Handle,
+                    time: 250,
+                    flags: NFWinUserAnimateWindowConstnats.AW_SLIDE | NFWinUserAnimateWindowConstnats.AW_VER_NEGATIVE
+                );
+            }
+            base.SetVisibleCore(value);
+        }
+
+        protected override void OnShown(EventArgs e)
+        {
+            base.OnShown(e);
+        }
+
+        private void SetWindowPosition()
+        {
+            int workingtop = Screen.PrimaryScreen.WorkingArea.Height - this.Height;
+            this.Top = workingtop - NotificationDialog.ScreenMargin;
+
+            int workingleft = Screen.PrimaryScreen.WorkingArea.Width - this.Width;
+            this.Left = workingleft - NotificationDialog.ScreenMargin;
+        }
+
+        private void ui_hide_window(object? sender, EventArgs? e)
```

Comment:
Nitpick: Naming convention is capitalised first letter title casing without underscores